app/server/models/exchange.py

```diff
@@ -40,7 +41,27 @@ class ExchangeContract(ModelBase):
         lazy=True, foreign_keys='TransferAccount.exchange_contract_id'
     )
 
-    def add_token(self, token):
+    def get_subexchange_details(self, token_address):
+        if self.subexchange_address_mapping is None:
+            return None
+        return self.subexchange_address_mapping.get(token_address, None)
+
+    def _add_subexchange(self, token_address, subexchange_address, subexchange_reserve_ratio_ppm):
+        if self.subexchange_address_mapping is None:
+            self.subexchange_address_mapping = {}
+        self.subexchange_address_mapping[token_address] = {
+            'subexchange_address': subexchange_address,
+            'subexchange_reserve_ratio_ppm': subexchange_reserve_ratio_ppm
```

Comment:
subexchanges are pegged at certain ratios? thought it'd fluctuate based on changes in the currency's different bonding curves, etc

Reply:
the `reserve_ratio` is what determines the extent of fluctuation with respect to changes in demand - i'll link the derivation in slack!